processor/schemaprocessor/internal/translation/revision_v1.go

```diff
@@ -6,109 +6,144 @@ package translation // import "github.com/open-telemetry/opentelemetry-collector
 import (
 	"go.opentelemetry.io/otel/schema/v1.0/ast"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/changelist"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/migrate"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/transformer"
 )
 
 // RevisionV1 represents all changes that are to be
 // applied to a signal at a given version.
+// todo(ankit) implement split and rest of otel schema
 type RevisionV1 struct {
-	ver              *Version
-	all              *migrate.AttributeChangeSetSlice
-	resource         *migrate.AttributeChangeSetSlice
-	spans            *migrate.ConditionalAttributeSetSlice
-	eventNames       *migrate.SignalNameChangeSlice
-	eventAttrsOnSpan *migrate.ConditionalAttributeSetSlice
-	eventAttrsOnName *migrate.ConditionalAttributeSetSlice
-	metricsAttrs     *migrate.ConditionalAttributeSetSlice
-	metricNames      *migrate.SignalNameChangeSlice
+	ver        *Version
+	all        *changelist.ChangeList
+	resources  *changelist.ChangeList
+	spans      *changelist.ChangeList
+	spanEvents *changelist.ChangeList
+	metrics    *changelist.ChangeList
+	logs       *changelist.ChangeList
```

Comment:
I like how this closely resembles the Schema File structure.

Reply:
Yeah it made it a lot easier to understand if I wrote it out this way